vscode/src/qirGeneration.ts

```diff
@@ -30,25 +32,43 @@ export async function getQirForActiveWindow(): Promise<string> {
     );
   }
 
-  // Check that the current target is base or adaptive_ri profile, and current doc has no errors.
   const targetProfile = getTarget();
-  if (targetProfile === "unrestricted") {
+  const is_unrestricted = targetProfile === "unrestricted";
+  const is_base = targetProfile === "base";
+
+  let error_msg =
+    supports_adaptive === undefined
+      ? "Generating QIR "
+      : "Submitting to Azure ";
```

Comment:
I'm assuming from the logic that the parameter is only passed when submitted to Azure? Maybe a comment to that effect.

Reply:
Yes, that's correct. I added some comments here and in the signature to clarify the intent.